modules/osdk-control-compat.adoc

```diff
@@ -0,0 +1,87 @@
+APIs that will be removed in the next OpenShift Container Platform link:https://access.redhat.com/support/policy/updates/openshift#ocp4_phases[Extended Update Support (EUS)] release.
+====
+
+To prevent workflow issues, cluster administrators must install a version of your Operator in their cluster that is compatible with the next version of {product-title} before they upgrade their cluster from any previous version. While it is recommended that you update your Operators to no longer use deprecated or removed APIs, if you still need to publish your Operator bundles with removed APIs for continued use on earlier versions of {product-title}, ensure that the bundle is configured accordingly.
+
+The following procedure helps prevent administrators from installing versions of your Operator on an incompatible version of {product-title}.  These steps also prevent administrators from upgrading to a newer version of {product-title} that would be incompatible with the version of your Operator that is currently installed on their cluster.
+
+[IMPORTANT]
+====
+You should configure the following settings in your Operator project as soon as possible. Users running workloads with deprecated APIs that are planning to eventually upgrade to a future version of {product-title}, which might remove the deprecated API, must be running an Operator that has the proper configuration to prevent the cluster upgrade and potentially adversely impact their critical workloads.
+====
+
+.Prerequisites
+
+- Operator SDK CLI installed on a development workstation
+- Operator project initialized by using the Operator SDK
+
+.Procedure
+
+. Configure the maximum version of {product-title} that your Operator is compatible with. In your Operator project's cluster service version (CSV), set the `olm.openShiftMaxVersion` annotation to prevent administrators from upgrading their cluster before upgrading the installed Operator to a compatible version:
++
+.Example CSV with `olm.maxOpenShiftVersion` annotation
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    "olm.properties": '[{"type": "olm.maxOpenShiftVersion", "value": "4.8"}]' <1>
+----
+<1> Setting `value` to `4.8` prevents cluster upgrades to {product-title} versions later than 4.8 when this bundle is installed on a cluster.
+
+. Configure the compatible versions of {product-title} for your Operator:
+
+.. Set the `com.redhat.openshift.versions` annotation in your project's `bundle/metadata/annotations.yaml` file:
++
+.Example `bundle/metadata/annotations.yaml` file with compatible versions
+[source,yaml]
+----
+com.redhat.openshift.versions: "v4.6-v4.8" <1>
+----
+<1> Set to a range or single version.
+
+.. To prevent your bundle from being carried on to an incompatible version of {product-title}, ensure that the index image is generated with the proper `com.redhat.openshift.versions` label in your project's `bundle.Dockerfile`:
```

Comment:
this is a bit weird to include, because this annotation is only consumed/enforced by the redhat operator pipelines when they assemble catalogs, as far as i know.  Someone building an operator that is distributed via their own catalog will not gain any benefit from this.  So if we are going to talk about it, i think we need to be more explicit about what it actually accomplishes.

cc @camilamacedo86

Reply:
HI @bparees, 

Thank you for your help on that. Based on our discussion in https://app.slack.com/client/T027F3GAJ/GHMALGJV6 and in the SDK docs:

> Bundle metadata in bundle/metadata/annotations.yaml contains information about a particular Operator version available in a registry. OLM uses this information to install specific Operator versions and resolve dependencies. That file and bundle.Dockerfile contain the same annotations, the latter as LABELs, which do not need to be modified in most cases; if you do decide to modify them, both sets of annotations **must** be the same to ensure consistent Operator deployment.

I understand that: 
- The correct place to add the info would be the annotations.yml (the bundle is the source)
- Unfortunately, for a lot of technical reasons in OLM runtime, we will use the bundle.Dockerfile labels and not the annotations
- Then, we have pipelines checking both (CVP) or only bundle.Dockerfle(IVs) or only the annotations(community) which generates the bundle.Dockerfile with the labels based on the info informed in the annotations. 

In this way, besides I agree that is not a good user experience I do not see how can we ask here for the user only ensures one place. Also, we have tracked an issue for SDK to ensure that will sync all in the `make bundle`, however, it is not done yet and will not be part of its downstream bin for 4.8. (https://github.com/operator-framework/operator-sdk/issues/5002)

WDYT?

Reply:
HI @bparees,

Thank you for your help on that. +1

We need to clarify that this step is only valid/useful/required if the bundle will be distributed in`Red Hat-provided Operator catalogs - https://deploy-preview-34166--osdocs.netlify.app/openshift-enterprise/latest/operators/understanding/olm-rh-catalogs.html

Reply:
@bparees @camilamacedo86 Updated step 2 with the following:

> If your bundle is intended for distribution in a Red Hat-provided Operator catalog, configure the compatible versions of OpenShift Container Platform for your Operator by setting the following properties:
> 
> NOTE: If your bundle is only intended for distribution in a custom catalog, you can skip this step.

https://deploy-preview-34166--osdocs.netlify.app/openshift-enterprise/latest/operators/operator_sdk/osdk-working-bundle-images#osdk-control-compat_osdk-working-bundle-images

Reply:
WDYT just we complement with something that makes more clear such as ;

>   If your bundle is only intended for distribution in a custom catalog, you can skip this step. The following step is only valid for when you publish your operator into Red Hat-provided Operator catalogs 

otherwise, it is lgtm for me.

Reply:
Updated NOTE to say:

> This step is only valid when publishing Operators in Red Hat-provided catalogs. If your bundle is only intended for distribution in a custom catalog, you can skip this step. For more details, see "Red Hat-provided Operator catalogs".